### DISCUSSION TITLE
[Feature]: Creating an Eslint plugin to flag large files in the pubic directory to save bills.

### DISCUSSION THREAD

**Discussion by Gitstar-OC** [vercel/next.js#73396] [Ideas]
### Goals

1. Currently Vercel's bandwith is costly and most of the role in it is being played by the public directory due to the uploading of assets to the cdn. This will be solved with this feature.

### Background

I think vercel's UI and UX is great but it comes at a great cost, we should not let that cost go on to increase more and more as we are the leading framework for the frontend and set up the standards for this things. This feature along with some other features might give vercel more market share in the hosting providers. 

### Proposal

We can create an eslint plugin or something similar that will take the size of the assets that are in the public directory and will sort them. If there are assets which are greater than 10 or let's say 100 MB the deployment will be stopped immediately and will throw an error. This limit could be changed by the developer and it will be up to him. 

Yes, I would love to contribute and will require just a little bit of help.

**Comment by denis23x** [comment]
Hi @Gitstar-OC 

You can achieve that functional much easier. Take a look at here https://github.com/typicode/husky

Just write pre-commit hook (sh/bash script), which will check your folder size and throw error if needed.

**Comment by Gitstar-OC** [comment]
Hey @denis23x, thanks. I did not know that a tool like this already exists. Will check it out :)

**Comment by Gitstar-OC** [comment]
Thanks for replying this soon.

### Happy Coding
